platform/default/sqlite_cache.cpp

```diff
@@ -470,16 +473,27 @@ void SQLiteCache::Impl::refresh(const Resource& resource, Seconds expires) {
     }
 }
 
-std::shared_ptr<SQLiteCache> SharedSQLiteCache::get(const std::string &path) {
-    std::shared_ptr<SQLiteCache> temp = masterPtr.lock();
-    if (!temp) {
-        temp.reset(new SQLiteCache(path));
-        masterPtr = temp;
+std::shared_ptr<SQLiteCache> SQLiteCache::getShared(const std::string &path) {
+    static std::mutex sharedMutex;
```

Comment:
Function level statics are initialized upon first execution, and there are no thread-safety guarantees in case of simultaneous first execution from multiple threads. This can be made 100% safe by moving the static to file level or using something like `pthread_once`.